src/data_table/local_data_table.js

```diff
@@ -459,6 +461,56 @@ class LocalDataTable extends View {
     this.dataTable.on("page", this._bulkCheckboxAdjust);
   }
 
+  _setupPaginationHistory() {
+    this.dataTable.on("page", function() {
+      var page = this.dataTable.fnPagingInfo().iPage;
+      if (page !== this._parseQueryString(window.location.search)) {
+        history.pushState({}, "pagination", this._createSearchString(this.dataTable.fnPagingInfo().iPage));
+      }
+    }.bind(this));
+    window.onpopstate = () => {
+      this._pageToSearchPage();
+    };
+    this.on("render", this._afterRender());
+  }
+
+  _afterRender() {
+    this._pageToSearchPage();
+  }
+
+  _pageToSearchPage() {
+    let pageNumber = this._parseQueryString(window.location.search);
+    if (pageNumber >= 0) {
+      this.page(this._parseQueryString(window.location.search));
+    }
+  }
+
+  _createSearchString(pageNumber) {
+    var queryString = window.location.search;
+    var searchString = "?";
+    pageNumber++;
+    var pageString = "page=" + pageNumber;
+    if (queryString[0] === '?') {
+      queryString = queryString.substr(1);
+      var parameters = $.deparam(queryString);
+      parameters.page = pageNumber;
+      searchString += $.param(parameters);
+    } else {
+      searchString += pageString;
+    }
+    return searchString;
+  }
+
+  _parseQueryString(search) {
+    if (search[0] === '?') {
+      search = search.substr(1);
+    }
+    let parameters = $.deparam(search);
+    let page = parseInt(parameters.page) - 1;
+    if (isNaN(page)) { return 0; }
+    return parseInt(parameters.page) - 1;
```

Comment:
for more clarity to the reader, i would put this into a full `if/else` so it's clear that NaN is a special case

Reply:
also this last line is just `page` right?  you already parsed and subtracted in L509.

```
let page = parseInt(parameters.page) - 1;
if (isNaN(page)) {
  return 0;
} else {
  return page;
}
```